docs/sources/migration-guide/migrating-from-thanos-or-prometheus.md

```diff
@@ -19,33 +19,41 @@ Each project stores blocks in different places and uses slightly different block
 
 For configuration of remote write to Grafana Mimir, refer to [Configuring Prometheus remote write]({{< relref "../operators-guide/secure/authentication-and-authorization.md#configuring-prometheus-remote-write" >}}).
 
-## Uploading historic blocks to the Grafana Mimir storage bucket
+## Uploading historic TSDB blocks to Grafana Mimir
+
+Grafana Mimir supports uploading of historic TSDB blocks, also from other systems such as Prometheus and Thanos.
+In order to enable this functionality, either for all tenants or a specific one, refer to
+[Configuring TSDB block upload]({{< relref "../operators-guide/configure/configuring-tsdb-block-upload.md" >}}).
 
 Prometheus stores TSDB blocks in the path specified in the `--storage.tsdb.path` flag.
 
-To find all blocks directories in the TSDB `<STORAGE TSDB PATH>`, run the following command:
+To find all block directories in the TSDB `<STORAGE TSDB PATH>`, run the following command:
 
 ```bash
 find <STORAGE TSDB PATH> -name chunks -exec dirname {} \;
 ```
 
-Grafana Mimir supports multiple tenants and stores blocks with a tenant prefix.
-With multi-tenancy disabled, there is a single tenant called `anonymous`.
-
-Copy each directory output from the previous command to the Grafana Mimir object storage bucket with
-your tenant prefix.
+Grafana Mimir supports multiple tenants and stores blocks per tenant. With multi-tenancy disabled, there
+is a single tenant called `anonymous`.
 
-- For AWS S3, use the `aws` tool in the following command:
+Use Grafana mimirtool to upload each block, e.g. identified by the previous command, to Grafana Mimir:
 
 ```bash
-aws s3 cp <DIRECTORY> s3://<TENANT>/<DIRECTORY>
+mimirtool backfill --user=<tenant> --address=http://<mimir-hostname> --id=<tenant> <block1> <block2>...
 ```
 
-- For Google Cloud Storage (GCS), use the `gsutil` tool in the following command:
+**Note**: If you need to authenticate against Grafana Mimir, you can provide an API key via the `--key` flag
+(e.g., `--key=$(cat token.txt)`).
 
-```bash
-gsutil -m cp -r <DIRECTORY> gs://<TENANT>/<DIRECTORY>
-```
+Grafana Mimir will perform some conversions of each block, if they are from another system
+(e.g. Prometheus or Thanos). Only one label in meta.json is kept: `__compactor_shard_id__`.
+The following deprecated labels are stripped:
+
+- `__org_id__`
+- `__ingester_id__`
+- `__shard_id__`
+
```

Comment:
This may be too much information.
- Only Thanos, Cortex or Mimir blocks can have labels. Prometheus blocks have no labels.
- Labels listed as "Deprecated" labels are really deprecated only by Mimir. Cortex still uses `__org_id__` or `__ingester_id__`.

I'd suggest to drop this entire section, and promote block upload only for Prometheus blocks. *ALL* Thanos block will be rejected by this API, since Thanos blocks *must* use external labels. (We can mention that below in "Considerations on Thanos specific features").